Include/internal/pycore_stackref.h

```diff
@@ -8,187 +8,210 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+#include "pycore_object_deferred.h"
+
 #include <stddef.h>
 
 typedef union {
     uintptr_t bits;
 } _PyStackRef;
 
-static const _PyStackRef Py_STACKREF_NULL = { .bits = 0 };
 
 #define Py_TAG_DEFERRED (1)
 
-// Gets a PyObject * from a _PyStackRef
-#if defined(Py_GIL_DISABLED)
-static inline PyObject *
-PyStackRef_Get(_PyStackRef tagged)
-{
-    PyObject *cleared = ((PyObject *)((tagged).bits & (~Py_TAG_DEFERRED)));
-    return cleared;
-}
+#define Py_TAG_PTR      (0)
+#define Py_TAG_BITS     (1)
+
+#ifdef Py_GIL_DISABLED
+    static const _PyStackRef Py_STACKREF_NULL = { .bits = 0 | Py_TAG_DEFERRED};
 #else
-#   define PyStackRef_Get(tagged) ((PyObject *)((tagged).bits))
+    static const _PyStackRef Py_STACKREF_NULL = { .bits = 0 };
 #endif
 
-// Converts a PyObject * to a PyStackRef, stealing the reference.
-#if defined(Py_GIL_DISABLED)
-static inline _PyStackRef
-_PyStackRef_StealRef(PyObject *obj)
-{
-    // Make sure we don't take an already tagged value.
-    assert(((uintptr_t)obj & Py_TAG_DEFERRED) == 0);
-    return ((_PyStackRef){.bits = ((uintptr_t)(obj))});
-}
-#   define PyStackRef_StealRef(obj) _PyStackRef_StealRef(_PyObject_CAST(obj))
+#define PyStackRef_IsNull(stackref) ((stackref).bits == Py_STACKREF_NULL.bits)
+
+
+#ifdef Py_GIL_DISABLED
+#   define PyStackRef_True ((_PyStackRef){.bits = ((uintptr_t)&_Py_TrueStruct) | Py_TAG_DEFERRED })
```

Comment:
By the way, you can use:

```c
static const _PyStackRef PyStackRef_True =  { (uintptr_t)&_Py_TrueStruct + Py_TAG_DEFERRED };
```

For some reason, `+` is okay, but bitwise `|` is not for constant expressions involving pointers.

Reply:
Mark tried that, but Windows rejects it.